schema/ksl-schema.kdl

```diff
@@ -0,0 +1,925 @@
+                """ {
+                arg {
+                    type number
+                }
+            }
+            node gte about="""
+                Greater than or equal to
+
+                Only used for numeric values. Constrains them to be greater than
+                or equal to the given number.
+                """ {
+                arg {
+                    type number
+                }
+            }
+            node lt about="""
+                Less than
+
+                Only used for numeric values. Constrains them to be less than
+                the given number.
+                """ {
+                arg {
+                    type number
+                }
+            }
+            node lte about="""
+                Less than or equal to
+
+                Only used for numeric values. Constrains them to be less than or
+                equal to the given number
+                """ {
+                arg {
+                    type number
+                }
+            }
+        }
+    }
+    // Validations shared across all types.
+    node shared-validations id=shared-validations {
+        children {
+            node type about="The type for this value\n\nMultiple arguments signify a sum type." {
+                repeatable
+                args {
+                    min 1
+                    type string
+                    enum string boolean number integer #null
+                    distinct
+                }
+            }
+            // TODO: establish equality expectations.
+            node const about="""
+                Specific value
+
+                Exact value that this value must match. Equivalent to a
+                single-value `enum` validation.
+                """ {
+                arg about="Constant value"
+            }
+            node enum about="""
+                Enumeration of values
+
+                An enumeration of possible values
+                """ {
+                repeatable
+                args about="Enumeration choices" {
+                    min 1
+                }
+                children {
+                    node allow-others about="""
+                        Allow other choices
+
+                        Whether other values than those explicitly enumerated
+                        may be provided, so long as they pass other validations
+                        in the node.
+
+                        While apparently redundant, this option may be useful in
+                        cases where there's a set of suggested values, but
+                        others are acceptable. This information can then be used
+                        by tooling to e.g. suggest completion items.
+                        """ {
+                        arg {
+                            type boolean
+                            default #true
+                        }
+                    }
+                    node - about="Enumeration choice" {
```

Comment:
A bit confused by this `-` here? I may have just missed it elsewhere?

Reply:
There's a convention in KDL about having homogenous child lists simply use `-` as the node name.